src/module.cpp

```diff
@@ -495,7 +503,7 @@ Expr *lCreateConstExpr(ExprList *exprList, const AtomicType::BasicType basicType
     } else {
         // T equals int8_t* and etc.
         using PointToType = typename std::remove_pointer<T>::type;
-        vals = (T)alloca(N * sizeof(PointToType));
+        vals = (T)ALLOCA(N * sizeof(PointToType));
```

Comment:
@nurmukhametov 
I think `alloca()` is not needed here. Why not T to be a set of `{bool, int8_t, ..., llvm::APVal}` instead of `{bool*, int8_t*, ..., std::vector<llvm::APVal>}` and inside of the function define `std::vector<T> vals`? That would simplify the logic and will allow getting rid of `alloca()`.

This should not block this PR though.

Reply:
> Why not T to be a set of {bool, int8_t, ..., llvm::APVal} instead of {bool*, int8_t*, ..., std::vector<llvm::APVal>} and inside of the function define std::vector<T> vals?

See comment here https://github.com/ispc/ispc/pull/2739#discussion_r1471150928

Reply:
You don't need vector here. `T vals[N];` should do the job.

Reply:
This will be fixed in separate PR as discussed.

Reply:
VLA won't work due to MSVC. With static big enough array `T vals[64]` there is another problem. `llvm::APFloat` doesn't have the default constructor, so I don't understand how to declare `llvm::APFloat vals[64]` without compilation error.

Reply:
use `std::array`?

Reply:
> use `std::array`?

Is there any difference?

```c++
#include <array>
#include "llvm/ADT/APFloat.h"

int main() {
    llvm::APFloat X[64];
    std::array<llvm::APFloat, 64> Y;
    return 0;
}
```

Both declarations are compilation errors due to missing constructor.

Reply:
My bad, I assumed that `std::array` may be initialized like `std::vector` from a single element. Looks like there's no elegant way to construct it on the stack.